jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -1844,9 +1844,10 @@ public boolean equalsIgnoreCase(String string) {
 				}
 				/*[ENDIF] JAVA_SPEC_VERSION >= 16 */
 
-				if (charAtO1 != charAtO2 &&
-						toUpperCase(charAtO1) != toUpperCase(charAtO2) &&
-						((charAtO1 <= 255 && charAtO2 <= 255) || Character.toLowerCase(charAtO1) != Character.toLowerCase(charAtO2))) {
+				if (charAtO1 != charAtO2
+						&& (Character.toLowerCase(toUpperCase(charAtO1)) != Character.toLowerCase(toUpperCase(charAtO2)))
+						&& (charAtO1 <= 255 && charAtO2 <= 255)
+				) {
```

Comment:
There are two `public boolean equalsIgnoreCase(String string)` within `String.java`, `Java 8` version seems missing.

Reply:
Please add the testcase supplied within the issue to `Test_String.java`.

Reply:
There is special handling in the code for the last character. When adding tests, it can test for the turkish i as the only character (which is also the last), and also when something else is the last character.